src/otp-input/otp-input.tsx

```diff
@@ -44,65 +51,54 @@ export const OtpInput = ({
     }, []);
 
     useEffect(() => {
-        if (!enableCountDownTimer || disabled) {
-            return;
+        if (countDown === cooldownDuration) {
+            if (onCooldownStart) onCooldownStart();
+            beginCountdown();
```

Comment:
return a clean up function to clear the interval

Reply:
do you mean something like this?
```
const beginCountdown() => {
        const interval = setInterval(....);
        
        return () => {
             clearInterval(interval);
        };
};
```

Then in `useEffect`

```
const cleanup = beginCountdown();

() => {
     cleanup();
};

```